packages/cli/src/evaluation.ee/test-runner/errors.ee.ts

```diff
@@ -12,16 +12,13 @@ export type TestCaseExecutionErrorCode =
 	| 'PAYLOAD_LIMIT_EXCEEDED'
 	| 'UNKNOWN_ERROR';
 
-export class TestCaseExecutionError extends Error {
+export class TestCaseExecutionError extends ApplicationError {
 	readonly code: TestCaseExecutionErrorCode;
 
-	readonly extra: IDataObject;
-
-	constructor(code: TestCaseExecutionErrorCode, extra: IDataObject = {}) {
-		super('Test Case execution: ' + code);
+	constructor(code: TestCaseExecutionErrorCode, extra: Record<string, unknown> = {}) {
+		super('Test Case execution: ' + code, { extra });
```

Comment:
Is this a good message? Could it be something like

`Test Case execution failed with code ${code}`?